pallets/phala/src/compute/stake_pool_v2.rs

```diff
@@ -0,0 +1,1478 @@
+				shares,
+				nft_id,
+				as_vault,
+				PoolType::StakePool,
+			)?;
+			nft_guard.save()?;
+			let _nft_id =
+				base_pool::Pallet::<T>::merge_or_init_nft_for_staker(pool_info.basepool.cid, who, pool_info.basepool.pid, PoolType::StakePool)?;
+			base_pool::pallet::Pools::<T>::insert(pid, PoolProxy::StakePool(pool_info.clone()));
+
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		#[frame_support::transactional]
+		pub fn migrate_stakepools(origin: OriginFor<T>, max_iterations: u32) -> DispatchResult {
+			let mut who = ensure_signed(origin)?;
+			base_pool::Pallet::<T>::ensure_migration_root(who)?;
+			let wrappedbalances_accountid = <T as wrapped_balances::Config>::WrappedBalancesAccountId::get();
+			let mut last_pid = StakepoolIterateStartPos::<T>::get();
+			let mut iter = match last_pid {
+				Some(pid) => {
+					let key: Vec<u8> = stake_pool::pallet::StakePools::<T>::hashed_key_for(pid);
+					stake_pool::pallet::StakePools::<T>::iter_from(key)
+				}
+				None => stake_pool::pallet::StakePools::<T>::iter(),
+			};
+			let mut i = 0;
+			for (pid, pool_info) in iter.by_ref() {
+				let collection_id: CollectionId = base_pool::Pallet::<T>::consume_new_cid();
+				let symbol: BoundedVec<u8, <T as pallet_rmrk_core::Config>::CollectionSymbolLimit> =
+					format!("STAKEPOOL-{}", pid)
+						.as_bytes()
+						.to_vec()
+						.try_into()
+						.expect("create a bvec from string should never fail; qed.");
+				pallet_rmrk_core::Pallet::<T>::create_collection(
+					Origin::<T>::Signed(base_pool::pallet_id::<T::AccountId>()).into(),
+					collection_id,
+					Default::default(),
+					None,
+					symbol,
+				)
+				.expect("create collection should not fail; qed.");
+				let account_id = base_pool::pallet::generate_staker_account::<T::AccountId>(
+					pid,
+					pool_info.owner.clone(),
+				);
+				let (owner_reward_account, lock_account) =
+					generate_owner_and_lock_account::<T::AccountId>(pid, pool_info.owner.clone());
+				let mut new_pool_info = StakePool {
+					basepool: base_pool::BasePool {
+						pid,
+						owner: pool_info.owner,
+						total_shares: pool_info.total_shares,
+						total_value: pool_info.total_stake,
+						withdraw_queue: VecDeque::new(),
+						value_subscribers: VecDeque::new(),
+						cid: collection_id,
+						pool_account_id: account_id,
+					},
+					payout_commission: pool_info.payout_commission,
+					cap: pool_info.cap,
+					workers: VecDeque::new(),
+					cd_workers: VecDeque::new(),
+					lock_account,
+					owner_reward_account,
+				};
+				pool_info.workers.into_iter().for_each(|pubkey| {
+					new_pool_info.workers.push_back(pubkey);
+					let session: T::AccountId = pool_sub_account(pid, &pubkey);
+					let worker_info = computation::pallet::Sessions::<T>::get(&session)
+						.expect("session data should exist; qed.");
+					if worker_info.state == computation::pallet::WorkerState::WorkerCoolingDown {
+						new_pool_info.cd_workers.push_back(pubkey);
+					}
+					let computing_stake =
```

Comment:
If it's zero, we don't need to mint

Reply:
I think it will not be a problem if we mint zero, and also the stakes in computation will never be zero i think

Reply:
done